include/zim/suggestion_iterator.h

```diff
@@ -31,11 +31,16 @@ class SuggestionResultSet;
 class SuggestionItem;
 class SearchIterator;
 
-class LIBZIM_API SuggestionIterator : public std::iterator<std::bidirectional_iterator_tag, SuggestionItem>
+class LIBZIM_API SuggestionIterator
 {
     typedef Archive::iterator<EntryOrder::titleOrder> RangeIterator;
     friend class SuggestionResultSet;
     public:
+        using iterator_category = std::bidirectional_iterator_tag;
+        using value_type = SuggestionItem;
+        using pointer = SuggestionItem*;
+        using referenc = SuggestionItem&;
```

Comment:
Typo (which kind of illustrates that no code depends on `SuggestionIterator::reference` :wink:)